src/Bicep.Core/Semantics/Decorator.cs

```diff
@@ -63,7 +67,14 @@ public void Validate(DecoratorSyntax decoratorSyntax, TypeSymbol targetType, ITy
                 return targetObject;
             }
 
-            return this.evaluator(functionCall, targetType, targetObject);
+            return this.evaluator(functionCall, RemoveImplicitNull(targetType), targetObject);
         }
+
+        private static TypeSymbol RemoveImplicitNull(TypeSymbol type) => type switch
+        {
+            UnionType union when union.Members.Where(m => !ReferenceEquals(m.Type, LanguageConstants.Null)).ToArray() is { } sansNull
+                && sansNull.Length < union.Members.Length => TypeHelper.CreateTypeUnion(sansNull),
+            _ => type,
+        };
```

Comment:
[nit] Could we make this into a reusable method?

Reply:
A reusable method got added in #9585 after this section was written. Updated to use `TypeHelper.TryRemoveNullability` instead.